### PR TITLE
🤖 ✅ Am I Real: Tests for analytics service

### LINKED ISSUE #2613: Analytics tests

### PR DESCRIPTION
**Pull request by greg-nagy**
Closes #2613

## Background
- [Illustration of the concept](https://www.youtube.com/watch?v=GJfsbhJY8gk)
- https://www.scientificamerican.com/article/confirmed-we-live-in-a-simulation/
- [https://www.sciencedirect.com/science/article/pii/S2090123221001491](https://www.youtube.com/watch?v=eBGIQ7ZuuiU)

Latest build: [extension-builds-2740](https://github.com/tallyhowallet/extension/suites/9894267491/artifacts/477918234) (as of Thu, 15 Dec 2022 19:15:09 GMT).

### CODE REVIEW

File: background/services/analytics/tests/index.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  beforeAll(() => {
+    global.fetch = jest.fn()
+    // We need this set otherwise the posthog lib won't send the events
+    process.env.POSTHOG_API_KEY = "hey hey hey"
+  })
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    analyticsService = await createAnalyticsService()
+    preferenceService = analyticsService["preferenceService"]
```
Comment (jagodarybacka) [inline background/services/analytics/tests/index.integration.test.ts:32]: if you are creating these services before each test don't you have to remove them after each one?

File: background/services/analytics/tests/index.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      )
+      expect(preferenceService.updateAnalyticsPreferences).toBeCalledTimes(1)
+    })
+
+    it("should generate a new uuid and save it to database", async () => {
+      // Called once for generating the new user uuid
+      // and once for the 'Background start' and once for the `New install' event
+      expect(uuid.v4).toBeCalledTimes(3)
+
+      expect(analyticsService["db"].setAnalyticsUUID).toBeCalledTimes(1)
```
Comment (jagodarybacka) [inline background/services/analytics/tests/index.integration.test.ts:136]: I would check if a new entry was added to the DB as this function can be called but updating DB can fail

File: background/services/analytics/tests/index.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+      jest.spyOn(preferenceService, "getAnalyticsPreferences")
+
+      expect(await preferenceService.getAnalyticsPreferences()).toEqual({
+        isEnabled: false,
+        hasDefaultOnBeenTurnedOn: false,
+      })
+
+      expect(preferenceService.getAnalyticsPreferences).toBeCalled()
+    })
+    it("should change the isEnabled output based on the changed feature flag", () => {
```
Comment (jagodarybacka) [inline background/services/analytics/tests/index.integration.test.ts:45]: Idk if changing flag tests should be written for each feature like this in the integration tests file. It won't hurt us but it feels to me like it's not a place for this test.

**Review by jagodarybacka — COMMENTED** [review]
Great PR 😍

File: background/services/analytics/tests/index.integration.test.ts (hunk 2 above)
Comment (greg-nagy) [reply background/services/analytics/tests/index.integration.test.ts:136]: Checking db directly felt out of depth for this integration test, but happy to do it if you feel strongly.

File: background/services/analytics/tests/index.integration.test.ts (hunk 1 above)
Comment (greg-nagy) [reply background/services/analytics/tests/index.integration.test.ts:32]: After every creation, the old service should be garbage collected.

**Pull request merged**